appserver/admin/cli-optional/src/main/java/com/sun/enterprise/admin/cli/optional/DerbyControl.java

```diff
@@ -37,214 +37,100 @@
- *  invoking NetworkServerControl from the StartDatabaseCommand
- *  class is so that a separate JVM is launched when starting the
- *  database and the control is return to CLI.  
- *  @author <a href="mailto:jane.young@sun.com">Jane Young</a>
- *  @version  $Revision: 1.13 $
+ * This class uses Java reflection to invoke Derby NetworkServerControl class.
+ * This class is used to start/stop/ping derby database. The reason for creating
+ * this class instead of directly invoking NetworkServerControl from the
+ * StartDatabaseCommand class is so that a separate JVM is launched when
+ * starting the database and the control is return to CLI.
+ *
+ * @author <a href="mailto:jane.young@sun.com">Jane Young</a>
+ * @version $Revision: 1.13 $
  */
-public final class DerbyControl
-{
-    final public static String DB_LOG_FILENAME = "derby.log";
-    final private String derbyCommand;
-    final private String derbyHost;
-    final private String derbyPort;
-    final private String derbyHome;
-    final private boolean redirect;
-    final private String derbyUser;
-    final private String derbyPassword;
+public final class DerbyControl extends DBControl {
 
+    public static final String DB_LOG_FILENAME = "derby.log";
 
-        //constructor 
     public DerbyControl(final String dc, final String dht, final String dp,
-                        final String redirect, final String dhe, final String duser, final String dpwd)
-    {
-        this.derbyCommand = dc;
-        this.derbyHost = dht;
-        this.derbyPort = dp;
-        this.derbyHome = dhe;
-	this.redirect = Boolean.valueOf(redirect).booleanValue();
-        this.derbyUser = duser;
-        this.derbyPassword = dpwd;
-
-	    if (this.redirect) {
+            final String redirect, final String dhe, final String duser, final String dpwd) {
+        super(dc, dht, dp, redirect, dhe, duser, dpwd);
 
-	        try {
-                String dbLog = "";
-                if (this.derbyHome == null) {
-                    // if derbyHome is null then redirect the output to a temporary file
-                    // which gets deleted after the jvm exists.
-		            dbLog = createTempLogFile();
-                }
-		    else {
-		        dbLog = createDBLog(this.derbyHome);
-		    }
-            
-                //redirect stdout and stderr to a file
-                PrintStream printStream = new PrintStream(new FileOutputStream(dbLog, true), true);
-                System.setOut(printStream);
-                System.setErr(printStream);
-            }
-            catch (Throwable t) {
-	            t.printStackTrace();
-	            //exit with an error code of 2
-	            Runtime.getRuntime().exit(2);
-            }
-        }
-        //do not set derby.system.home if derbyHome is empty
-        if (this.derbyHome!=null && this.derbyHome.length()>0) {
-            System.setProperty("derby.system.home", this.derbyHome);
+        //do not set derby.system.home if dbHome is empty
```

Comment:
Really, really small thing, but it's ever so slightly more readable when adding a space and starting with a capital:

```java
// Do not set derby.system.home if dbHome is empty
```